pymc/tests/test_sampling.py

```diff
@@ -1023,6 +1024,110 @@ def test_deterministics_out_of_idata(self, multitrace):
             ppc = pm.sample_posterior_predictive(trace, var_names="c", return_inferencedata=True)
         assert np.all(np.abs(ppc.posterior_predictive.c + 4) <= 0.1)
 
+    def test_logging_sampled_basic_rvs_prior(self, caplog):
+        with pm.Model() as m:
+            x = pm.Normal("x")
+            y = pm.Deterministic("y", x + 1)
+            z = pm.Normal("z", y, observed=0)
+
+        with m:
+            pm.sample_prior_predictive(samples=1)
+        assert caplog.record_tuples == [("pymc", logging.INFO, "Sampling: [x, z]")]
+        caplog.clear()
+
+        with m:
+            pm.sample_prior_predictive(samples=1, var_names=["x"])
+        assert caplog.record_tuples == [("pymc", logging.INFO, "Sampling: [x]")]
+        caplog.clear()
+
+    def test_logging_sampled_basic_rvs_posterior(self, caplog):
+        with pm.Model() as m:
+            x = pm.Normal("x")
+            x_det = pm.Deterministic("x_det", x + 1)
+            y = pm.Normal("y", x_det)
+            z = pm.Normal("z", y, observed=0)
+
+        idata = az_from_dict(posterior={"x": np.zeros(5), "x_det": np.ones(5), "y": np.ones(5)})
+        with m:
+            pm.sample_posterior_predictive(idata)
+        assert caplog.record_tuples == [("pymc", logging.INFO, "Sampling: [z]")]
+        caplog.clear()
+
+        with m:
+            pm.sample_posterior_predictive(idata, var_names=["y", "z"])
+        assert caplog.record_tuples == [("pymc", logging.INFO, "Sampling: [y, z]")]
+        caplog.clear()
+
+        # Resampling `x` will force resampling of `y`, even if it is in trace
+        with m:
+            pm.sample_posterior_predictive(idata, var_names=["x", "z"])
+        assert caplog.record_tuples == [("pymc", logging.INFO, "Sampling: [x, y, z]")]
+        caplog.clear()
+
+        # Missing deterministic `x_det` does not show in the log, even if it is being
+        # recomputed, only `y` RV shows
+        idata = az_from_dict(posterior={"x": np.zeros(5)})
+        with m:
+            pm.sample_posterior_predictive(idata)
+        assert caplog.record_tuples == [("pymc", logging.INFO, "Sampling: [y, z]")]
+        caplog.clear()
+
+        # Missing deterministic `x_det` does not cause recomputation of downstream `y` RV
+        idata = az_from_dict(posterior={"x": np.zeros(5), "y": np.ones(5)})
+        with m:
+            pm.sample_posterior_predictive(idata)
+        assert caplog.record_tuples == [("pymc", logging.INFO, "Sampling: [z]")]
+        caplog.clear()
+
+        # Missing `x` cousse downstream of `y` RV, even if it is present in trace
```

Comment:
Not sure what word you wanted to put here instead of cousse

Reply:
`cousse` is a French word for I had an aneurysm.

Reply:
Oh la la!

Reply:
I meant `causes sampling of downstream RV y` :P

Reply:
https://youtu.be/JhbJnlIvfyc

Reply:
I couldn’t resist the Simpson moment of this thread

Reply:
That's a great sketch. Didn't know it.